projects/igniteui-angular/src/lib/grids/column.component.ts

```diff
@@ -286,24 +287,23 @@ export class IgxColumnComponent implements AfterContentInit {
      */
     public set width(value: string) {
         if (value) {
+            this._calcWidth = null;
             this.widthSetByUser = true;
             this._width = value;
         }
     }
 
     public get calcWidth(): any {
-        const colWidth = this.width;
-        const isPercentageWidth = colWidth && typeof colWidth === 'string' && colWidth.indexOf('%') !== -1;
-        if (isPercentageWidth) {
-            return parseInt(colWidth, 10) / 100 * this.grid.unpinnedWidth;
-        } else if (!colWidth) {
-            // no width
-            return this.defaultWidth || this.grid.getPossibleColumnWidth();
-        } else {
-            return this.width;
+        if (this._calcWidth !== null && !isNaN(this.calcPixelWidth)) {
+            return this._calcWidth;
         }
+        this.cacheCalcWidth();
+        return this._calcWidth;
     }
 
+    private _calcWidth = null;
+    public calcPixelWidth: number;
```

Comment:
Is there a reason to introduce another public prop (calcPixelWidth), which has the same value as calcWidth but parsed to number? Having both seems a bit redundant since the igxForOf parses the values to number anyway.

Reply:
I want it cached, because the parseInt operation which is done on changes when calculating sizes is expensive.